pkg/core/consensus/selection/selector.go

```diff
@@ -138,30 +150,28 @@ func (s *Selector) IncreaseTimeOut() {
 	s.timeout = s.timeout * 2
 }
 
-func (s *Selector) publishBestEvent() error {
+func (s *Selector) sendBestEvent() error {
+	var bestEvent consensus.InternalPacket
 	s.eventPlayer.Pause(s.scoreID)
-	buf := new(bytes.Buffer)
 	s.lock.RLock()
-	bestEvent := s.bestEvent
+	bestEvent = s.bestEvent
 	s.lock.RUnlock()
+
 	// If we had no best event, we should send an empty hash
 	if bestEvent == nil {
```

Comment:
As we set the `bestEvent` to an `EmptyScore` when starting selection, this condition will never be true, and instead the selector will send a `bestScore` with everything zero'd out, which causes consensus to stall.

Reply:
I think this was an oversight from my side. I introduced the `Score.IsEmpty` method appositely as my intention is to get rid of as many `nil` check as we can. Please double check if changing to the following would implement the intended behavior:
```golang
if bestEvent.IsEmpty() {
    bestEvent = s.signer.Compose(emptyScoreFactory{})
}
```

Reply:
Yep, that should be fine, just tested it on the harness. (Although did need to typecast `bestEvent`)